Mathlib/Data/List/Basic.lean

```diff
@@ -230,6 +230,18 @@ theorem replicate_left_injective (a : α) : Injective (replicate · a) :=
 theorem replicate_left_inj {a : α} {n m : ℕ} : replicate n a = replicate m a ↔ n = m :=
   (replicate_left_injective a).eq_iff
 
+theorem head?_flatten_replicate {n : ℕ} {as : List α} (h : n ≠ 0) :
+((List.replicate n as).flatten.head?) = as.head? := by
+  match n, as with
+  | 0, _ => exact (h rfl).elim
+  | n+1, [] => simp
+  | n+1, h::_ => simp [replicate]
```

Comment:
```suggestion
  | n+1, h :: _ => simp [replicate]
```
I think (?) the convention is to have spaces on either side of the `::`, but happy to be overridden on this:)

Reply:
(feel free to ignore!) If you want a slightly shorter proof, you can also go for 
```
  obtain ⟨n, rfl⟩ := Nat.exists_eq_succ_of_ne_zero hn
  induction as <;> simp [replicate]
```